scripts/github_api

```diff
@@ -71,6 +71,40 @@ def _print_output(output):
         print '%s="%s"' % (k, v)
 
 
+class Status:
+    """
+    https://developer.github.com/v3/repos/statuses/
+    """
+    def __init__(self, repo, reference):
+        """
+        Args:
+            repo (str): target repository path. Format:
+                <repo owner>/<repo>
+            reference (str): git reference. Either a sha1 or a
+                pull-request number
```

Comment:
FR> yes, that's what I meant

Pls fix doc then

PR> I don't see the need to couple this to pull requests. Accepting a reference makes it more true to the API and allow this class to be used in other situations too.

1) I personally prefer a clearer OO model than trying to mimick 1-1 the underlying API. We have talked about this when you were working in the Blivet wrapper in PowerKVM.  2) IMHO the data representation would be clearer and easier to understand in the API wrapper if there was a 1-1 reference between Pull Request and PullRequestStatus. A GitHub status does not make sense without a pull request IMHO, what this API call with the git reference probably does is find the PR number from the git reference. 3) we do not have any need to create status from a git reference AFAIK.  4) *If* we need such feature in the future, we could simply create an util function (not inside any class) to do that.